template.js

```diff
@@ -34,16 +34,94 @@ if (data.type === 'track') {
     sendAliasRequest();
 } else if (data.type === 'identify') {
     sendIdentifyRequest();
+} else if (data.type === 'profile-set') {
+    sendSetProfileRequest();
+} else if (data.type === 'profile-append') {
+    sendAppendProfileRequest();
 } else if (data.type === 'reset') {
     cookieOptions['max-age'] = 1;
-
     setCookie('stape_mixpanel_distinct_id', 'empty', cookieOptions);
     setCookie('stape_mixpanel_device_id', 'empty', cookieOptions);
     data.gtmOnSuccess();
-
     return;
 }
 
+
+function sendAppendProfileRequest() {
+    // Construct the properties to append from the table field input
+    const propertiesToAppend = {};
+    data.userPropertiesToAppend.forEach(row => {
+        if (!propertiesToAppend[row.propertyName]) {
+            propertiesToAppend[row.propertyName] = [];
+        }
+        propertiesToAppend[row.propertyName].push(row.valueToAppend);
+    });
+
+    const profileBody = {
+        '$token': data.token,
+        '$distinct_id': getDistinctId(),
+        '$append': propertiesToAppend
+    };
+
+    const postUrl = 'https://' + (data.serverEU ? 'api-eu.mixpanel.com' : 'api.mixpanel.com') + '/engage#profile-list-append';
```

Comment:
Move domain define to global variable

Reply:
Ok, I moved it to the global property but still the rest of the functions use local ones, so they also should be moved, right?

Reply:
it will be great if you update all functions to use global variable. thanks.

Reply:
Sure, but I still have some issues with identifying a user properly. I'll update the code again when I resolve this since there might be some further modifications needed in this regard